### PR TITLE
solvers: Added tests for solve()

### LINKED ISSUE #6819: powsimp should be attempted for multiple generators case in solve
```
>>> var('a,b,c,d',positive=True)
(a, b, c, d)
>>> solve(a*b**x-c*d**x, x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\solvers\solvers.py", line 826, in solve
    solution = _solve(f[0], *symbols, **flags)
  File "sympy\solvers\solvers.py", line 1333, in _solve
    "\nNo algorithms are implemented to solve equation %s" % f)
NotImplementedError: multiple generators [b**x, d**x]
No algorithms are implemented to solve equation b**x

But solve does know how to solve this:

>>> solve(b**x-a/c,x)
[(log(a) - log(c))/log(b)]

And powsimp can reduce the ratio of powers to a single power:

>>> powsimp(b**(-x)*d**x)
(d/b)**x

So a solution should be possible. But the algorithm doesn't leave the power together:

>>> solve(_-a/c,x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\solvers\solvers.py", line 826, in solve
    solution = _solve(f[0], *symbols, **flags)
  File "sympy\solvers\solvers.py", line 1333, in _solve
    "\nNo algorithms are implemented to solve equation %s" % f)
NotImplementedError: multiple generators [b**x, d**x]
No algorithms are implemented to solve equation b**x
```

Original issue for #6819: http://code.google.com/p/sympy/issues/detail?id=3720
Original author: https://code.google.com/u/117933771799683895267/

### PR DESCRIPTION
**Pull request by evatiwari**
#### References to other Issues or PRs

Fixes #6819

#### Brief description of what is fixed or changed
Tests added for the case where solve is used for an expression involving exponential terms (b**x), and reducing
multiple such terms into a single term raised to x.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/tests/test_solvers.py [hunk 1]
```diff
@@ -2290,3 +2289,9 @@ def test_issue_19509():
         -d + a + sqrt(-b + c),
         d + a - sqrt(-b - c),
         d + a + sqrt(-b - c)]
+
+def test_issue_6819():
+    from sympy import var
+    x = symbols('x')
+    var('a, b, c, d', positive = True)
+    assert solve(a*b**x - c*d**x, x) == [log(c/a)/log(b/d)]
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_solvers.py:2297]: We shouldn't use `var` in the codebase or the tests. This should use `symbols`.

**Comment by smichr** [comment]
two nits: leave 2 lines before the `def` and no space around the `=` after a keyword: 'positive=`. Otherwise this looks fine.

**Comment by evatiwari** [comment]
sure thing, on it

**Comment by oscarbenjamin** [comment]
The changes looks good. Can you squash the commits as explained here: https://github.com/sympy/sympy/pull/20700#issuecomment-757067113

The commit message could be:
```
test(solvers): add regression test for GH-6819
```

**Comment by oscarbenjamin** [comment]
Looks like you've accidentally added the t.diff file in the commit and also deleted a newline in `test_solvers.py`. The commit will need to be redone:
```
git reset HEAD^
git status
git diff  # check the changes
# Add the newline back in test_solvers.py
git add sympy/solvers/tests/test_solvers.py
git commit
git push --force
```

**Comment by evatiwari** [comment]
@oscarbenjamin is ```git reset HEAD^``` a complete command? I keep getting ```More?```

**Comment by oscarbenjamin** [comment]
> is `git reset HEAD^` a complete command?

Yes, but don't run it more than once. Each time you run it a commit is removed.

**Comment by evatiwari** [comment]
@oscarbenjamin I am trying to add the new line back but it isn't showing up in the file changes on github

**Comment by oscargus** [comment]
Closing this as the corresponding test has been added elsewhere.

**Pull request closed without merging**